src/lib/storage/value_segment.cpp

```diff
@@ -6,68 +6,89 @@
 namespace opossum {
 
 template <typename T>
-ValueSegment<T>::ValueSegment(bool nullable) {
-  // Implementation goes here
-}
+ValueSegment<T>::ValueSegment(bool nullable) : _is_nullable{nullable} {}
 
 template <typename T>
 AllTypeVariant ValueSegment<T>::operator[](const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  if (is_null(chunk_offset)) {
+    return NULL_VALUE;
+  }
+  return _values.at(chunk_offset);
 }
 
 template <typename T>
 bool ValueSegment<T>::is_null(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  return _is_nullable && _null_values.at(chunk_offset);
 }
 
 template <typename T>
 T ValueSegment<T>::get(const ChunkOffset chunk_offset) const {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+  const auto optional = get_typed_value(chunk_offset);
+  if (!optional.has_value()) {
+    throw std::logic_error{"value is NULL"};
+  }
```

Comment:
Although it is not clear from the tests, we figured that the `std::logic_error` is usually raised by the `Fail` and `Assert` macros from the project utils:
```suggestion
  Assert(optional.has_value(), "value is NULL");
```